src/IndexingMsg.chpl

```diff
@@ -607,32 +607,56 @@ module IndexingMsg
             when (DType.BigInt, DType.BigInt) {
                 var e = toSymEntry(gEnt,bigint);
                 var val = valueArg.getBigIntValue();
-                if e.max_bits != -1 {
-                  mod(val, val, e.max_bits);
+                var max_size = 1:bigint;
+                var has_max_bits = e.max_bits != -1;
+                if has_max_bits {
+                    max_size <<= e.max_bits;
+                    max_size -= 1;
+                }
+                if has_max_bits {
+                    val &= max_size;
```

Comment:
we check the same condition twice in a row, we really only need one `if` block here

In the same vein, we don't really need to declare `has_max_bits` as a variable since it should really only ever be used once

Reply:
Updated